File: leetcode2269/leetcode2269.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int divisorSubstrings(int num, int k) {
            string s = to_string(num);
            int ans = 0;
            for (int i = k; i <= s.size(); i++){
                int x = stoi(s.substr(i - k, k));
                if (x > 0 && num % x == 0){
                    ans ++;
                }
            }
            return ans;
        }
    };

File: leetcode2269/leetcode2269.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)
        ans = 0
        for i in range(k, len(s) + 1):
            x = int(s[i - k: i])
            if x > 0 and num % x == 0:
                ans += 1
        return ans